ceph.spec.in

```diff
@@ -81,6 +81,19 @@
 %endif
 %endif
 
+%if 0%{?suse_version}
+%if 0%{?is_opensuse}
+%global luarocks_package lua53-luarocks
+%bcond_without lua_modules
+%else
+# SLE does not support luarocks
+%bcond_with lua_modules
+%endif
+%else
+%global luarocks_package luarocks
+%bcond_without lua_modules
+%endif
```

Comment:
This conditional makes my brain hurt. It seems like it's trying to say this:

```
%if 0%{?suse_version} && ! 0%{?is_opensuse}
# SLE does not support luarocks
%bcond_with lua_modules
%else
%global luarocks_package luarocks
%bcond_without lua_modules
%endif
```

or this:

```
%if ! 0%{?suse_version} || 0%{?is_opensuse}
%global luarocks_package luarocks
%bcond_without lua_modules
%else
# SLE does not support luarocks
%bcond_with lua_modules
%endif
```

Reply:
i mixed the package name logic in the same condition. but taking it away this should be:
```
 %if 0%{?suse_version} && 0%{?is_opensuse}
￼%bcond_without lua_modules
￼%else
￼# SLE does not support luarocks
￼%bcond_with lua_modules
￼%endif
```
it also puzzles me...  but  setting: `￼%bcond_without lua_modules` means that if you check `%if 0%{with lua_modules}` it would return `true`

Reply:
@yuvalif I am suggesting to replace the complicated nested conditional (with two `%else` clauses) with a single, easy-to-understand conditional (with just one `%else`) clause. The conditional you wrote would not replace the entire conditional.

And yes, the meaning of `%bcond_with` and `%bcond_without` is clear, though it certainly can be puzzling when one is faced with it for the first time.

Reply:
In other words, I'm suggesting to replace this nested conditional:

```
%if 0%{?suse_version}
%if 0%{?is_opensuse}
%global luarocks_package lua53-luarocks
%bcond_without lua_modules
%else
# SLE does not support luarocks
%bcond_with lua_modules
%endif
%else
%global luarocks_package luarocks
%bcond_without lua_modules
%endif
```

with one of the two simple variants I suggested - both of which are, I believe, logically equivalent to the nested conditional.

Reply:
@smithfarm how about the following:

```
%if 0%{?suse_version} && !0%{?is_opensuse}
# SLE does not support luarocks
%bcond_with lua_modules
%elif 0%{?suse_version} && 0%{?is_opensuse}
%global luarocks_package lua53-luarocks
%bcond_without lua_modules
%else
%global luarocks_package luarocks
%bcond_without lua_modules
%endif
```

Reply:
:shrug: Even that would be an improvement :-)

Reply:
"elif" was just recently implemented in rpm, and does not work on opensuse.
have to use nested if.